things/api.py

```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*-
+
+"""
+Module implementing Things API
+"""
+
+
+from .database import Database
+
+
+def tasks(type="task", status="open", area=None, project=None, heading=None, **kwargs):
+    """
+    Read tasks into a list of dicts.
```

Comment:
what about starting now with a unit test that uses the demo database and expects e.g. a certain number of tasks?

Reply:
I don't know about you, but I don't feel very certain about this API quite yet. I'd enjoy to play around a bit and see if this design makes sense in practice. What do you think?

Reply:
I think to get a feeling about an API, a test-driven approach makes sense ;)
In https://github.com/thingsapi/things.py/commit/527c080f63c91c28b7973c6713c81ba7cbae71d8 I just added a few tests that cover 96% of the API and 73% of the database, however, without checking if the results make sense.

Reply:
👌 
I'm less practiced with the test-driven approach.